xilem/examples/stopwatch.rs

```diff
@@ -0,0 +1,261 @@
+        self.completed_lap_splits
+            .push(split_end.sub(self.split_start_time));
+        self.split_start_time = split_end;
+    }
+
+    fn get_current_duration(&self) -> Duration {
+        match self.last_start_time {
+            Some(last_start_time) => match last_start_time.elapsed() {
+                Ok(elapsed) => self.added_duration + elapsed,
+                Err(err) => {
+                    warn!("error calculating elapsed time: {}", err.to_string());
+                    self.added_duration
+                }
+            },
+            _ => self.added_duration,
+        }
+    }
+
+    fn update_display(&mut self) {
+        self.displayed_duration = self.get_current_duration();
+    }
+}
+
+fn get_formatted_duration(dur: &Duration) -> String {
+    let seconds = dur.as_secs_f64() % 60.0;
+    let minutes = (dur.as_secs() / 60) % 60;
+    let hours = (dur.as_secs() / 60) / 60;
+    format!("{hours}:{minutes:0>2}:{seconds:0>4.1}")
+}
+
+fn app_logic(data: &mut Stopwatch) -> impl WidgetView<Stopwatch> {
+    fork(
+        flex((
+            FlexSpacer::Fixed(5.0),
+            label(get_formatted_duration(&data.displayed_duration)).text_size(70.0),
+            flex((lap_reset_button(data), start_stop_button(data))).direction(Axis::Horizontal),
+            FlexSpacer::Fixed(1.0),
+            laps_section(data),
+            label(data.displayed_error.as_ref()),
+        )),
+        data.active.then(|| {
+            // Only update while active.
+            async_repeat(
+                |proxy| async move {
+                    let mut interval = time::interval(Duration::from_millis(50));
+                    loop {
+                        interval.tick().await;
+                        let Ok(()) = proxy.message(()) else {
+                            break;
+                        };
+                    }
+                },
+                |data: &mut Stopwatch, ()| {
+                    data.update_display();
+                },
+            )
+        }),
+    )
+}
+
+/// Creates a list of items that shows the lap number, split time, and total cumulative time.
+fn laps_section(data: &mut Stopwatch) -> Vec<AnyFlexChild<Stopwatch>> {
+    let mut items: Vec<AnyFlexChild<Stopwatch>> = Vec::new();
+    let mut total_dur = Duration::ZERO;
+    let current_lap = data.completed_lap_splits.len();
+
+    for (i, split_dur) in data.completed_lap_splits.iter().enumerate() {
+        total_dur = total_dur.add(*split_dur);
+        items.push(single_lap(i, split_dur, &total_dur).into_any_flex());
+    }
+    let current_split_duration = data.get_current_duration().sub(total_dur);
+    let mut reversed = Vec::new();
+    reversed.push(
+        single_lap(
+            current_lap,
+            &current_split_duration,
+            &data.get_current_duration(),
+        )
+        .into_any_flex(),
+    );
+    for item in items.into_iter().rev() {
+        reversed.push(item);
+    }
+    reversed
+}
```

Comment:
```suggestion
fn laps_section(data: &mut Stopwatch) -> FlexSequence<Stopwatch> {
    let mut items = Vec::new();
    let mut total_dur = Duration::ZERO;
    let current_lap = data.completed_lap_splits.len();

    for (i, split_dur) in data.completed_lap_splits.iter().enumerate() {
        total_dur = total_dur.add(*split_dur);
        items.push(single_lap(i, split_dur, &total_dur));
    }
    let current_split_duration = data.get_current_duration().sub(total_dur);
    let mut reversed = Vec::new();
    reversed.push(single_lap(
        current_lap,
        &current_split_duration,
        &data.get_current_duration(),
    ));
    for item in items.into_iter().rev() {
        reversed.push(item);
    }
    reversed
}
```

So this is what I mean, I have tried `impl WidgetViewSequence<Stopwatch> `, but I think it's incompatible in contexts requiring a `FlexSequence` (the `SuperElement` magic only seems to work I think when it's not yet explicitely typed yet, so inference can kick in). So this above would work instead, or alternatively also `Vec<impl WidgetView<Stopwatch>>` as return type.

(Requires imports obviously)

Reply:
That works. Thanks! Rust's typing is a little unusual compared to what I'm used to.